November Release Blogpost (#231)
conda/conda-dot-org: blog/2024-12-05-november-releases.md

---
title: "November 2024 Releases"
slug: "2024-12-05-november-releases"
authors: [kenodegard]
tags: [announcement, conda, conda-build, conda-libmamba-solver]
description: |
  conda 24.11.0, conda-build 24.11.2, and conda-libmamba-solver 24.11.1 have been released! 🎉
image: img/blog/2024-12-05-november-releases/banner.png
---

The November 2024 releases included updates to conda, conda-build, and conda-libmamba-solver! 🎉 All of these have been released to both `main` and `conda-forge` channels.

<!-- truncate -->

## Changes in conda [24.11.0](https://github.com/conda/conda/releases/tag/24.11.0)

To update `conda` to the latest version, run:

```bash
conda install -n base conda=24.11.0
```

### ✨ What's New? ✨

> - Adds a new plugin hook for reporter backends.

Author: ```suggestion
- Add a new plugin hook for reporter backends for customizing conda's output.
```

> - Add support for CEP-17 that allows specifying the path to the site-packages directory of the the Python package via the `repodata.json`.

Reviewer: Add / Adds tense inconsistency

Reviewer 2: +1 on this. I believe that we usually go with "add" over "adds" (or the equivalent).

> - Adds progress bar support for reporter backends plugin hook.
> - Adds support for defining spinners for the reporter backends plugin hook.
> - Adds support for confirmation functions for reporter backends plugin hook.

Author: ```suggestion
- Add progress bar support for reporter backends plugin hook.
- Add support for defining spinners for the reporter backends plugin hook.
- Add support for confirmation functions for reporter backends plugin hook.
```

- Fix output writing for `conda export --json --file`.
- Update `deprecated.action()` function to account for positional arguments that have no value specified.
- Fix continuous integration upload of coverage files.

### 🌅 What Got Deprecated or Got Marked for Future [Deprecation](https://github.com/conda-incubator/ceps/blob/main/cep-9.md)? 🌅

- Remove `__conda_reactivate` shell function in favor of `__conda_activate reactivate`.
- Mark `conda.misc.rel_path` as pending deprecation.
- Require Python 3.9 or greater.

## Changes in conda-build [24.11.0](https://github.com/conda/conda-build/releases/tag/24.11.0)/[24.11.1](https://github.com/conda/conda-build/releases/tag/24.11.1)/[24.11.2](https://github.com/conda/conda-build/releases/tag/24.11.2)

To update `conda-build` to the latest version, run:

```bash
conda install -n base conda-build=24.11.2
```

### ✨ What's New? ✨

- Introduce `--package-format` as a command line argument.
  - This takes precedence over default value and `condarc`.
  - Normalization occurs so `1`, `"1"`, `tar.bz2`,`.tar.bz2`, `2`, `"2"`, `conda`, `.conda` are all recognized and mapped appropriately.
  - Other options are rejected.

> - Add  support for [CEP-17](https://github.com/conda/ceps/blob/main/cep-0017.md) that allows specifying the location of the site-packages directory with the `python_site_packages_path` build option for any packages named `python`.

Author: ```suggestion
- Add support for [CEP-17](https://github.com/conda/ceps/blob/main/cep-0017.md) that allows specifying the location of the site-packages directory with the `python_site_packages_path` build option for any packages named `python`.
```

### 🔧 What Got Fixed? 🔧

- Fix regex for Jinja2 `set` / `for` statements to be more specific.
- Fix `ruamel.yaml` usage to use supported APIs.

> - Fix bug variant variables were not defined for the first parsing pass of a recipe.

Reviewer 2: ```suggestion
- Fix a bug where variant variables were not defined for the first parsing pass of a recipe.
```
For consistency with the two below

- Fix a bug where variants were incorrectly found as being used when they matched a leading substring of another variant.
- Fix a bug where variants were not found when variables were used in `pin_*` statements.
- (24.11.1) Fix `TypeError` when no CLI arguments are passed.

> - (24.11.2) Fix bug where `.copy()` was used on a string instead of `copy()` when processing variants.

Reviewer 2: ```suggestion
- (24.11.2) Fix a bug where `.copy()` was used on a string instead of `copy()` when processing variants.
```
